docs(adev): Document fake async tests in zoneless mode, with vitest and jasmine

aparzi opened the pull request:
Add comprehensive guide for fake async tests in zoneless mode, with vitest and jasmine

Fixes #64962 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:

Issue Number: #64962 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

atscott commented on adev/src/content/guide/testing/components-scenarios.md:
````diff
@@ -261,6 +261,782 @@ IMPORTANT: Be aware that asynchronous tasks that happen inside the `fakeAsync` z
+}
+```
+
+#### Testing with Angular's `fakeAsync()` (Jasmine)
````
Don't we already have documentation on `fakeAsync`? https://angular.dev/guide/testing/components-scenarios#async-test-with-fakeasync

atscott commented on adev/src/content/guide/testing/components-scenarios.md:
````diff
@@ -261,6 +261,782 @@ IMPORTANT: Be aware that asynchronous tasks that happen inside the `fakeAsync` z
+});
+```
+
+---
````
This is a pretty verbose way of showing very minor differences between mock clock implementations.

atscott reviewed (changes requested):
I don't think this is the right direction for what we should document. We should not create verbose documentation that compares different APIs from different test frameworks. Angular documentation on testing should document testing as it relates specifically to Angular's testing APIs. Testing timeouts inside a component is not different from testing timeouts in plain javascript. The same goes for rxjs operators. 

It definitely looks like a lot of work was put into this and I appreciate the effort here, but I don't personally feel like this is the right approach.

aparzi commented:
> I don't think this is the right direction for what we should document. We should not create verbose documentation that compares different APIs from different test frameworks. Angular documentation on testing should document testing as it relates specifically to Angular's testing APIs. Testing timeouts inside a component is not different from testing timeouts in plain javascript. The same goes for rxjs operators.
> 
> It definitely looks like a lot of work was put into this and I appreciate the effort here, but I don't personally feel like this is the right approach.

I'll try to streamline what I wrote, starting from the feedback you left me.

aparzi committed:
docs(adev): Document fake async tests in zoneless mode, with vitest and jasmine

Add comprehensive guide for fake async tests in zoneless mode, with vitest and jasmine

Fixes #64962

aparzi commented:
> I don't think this is the right direction for what we should document. We should not create verbose documentation that compares different APIs from different test frameworks. Angular documentation on testing should document testing as it relates specifically to Angular's testing APIs. Testing timeouts inside a component is not different from testing timeouts in plain javascript. The same goes for rxjs operators.
> 
> It definitely looks like a lot of work was put into this and I appreciate the effort here, but I don't personally feel like this is the right approach.

Thank you for the clear feedback. You're right that the previous approach was too verbose and framework-comparative. I've streamlined the documentation to focus specifically on Angular's testing APIs (fakeAsync(), tick(), whenStable()) and how they interact with common async scenarios in Angular components.
I'm available for any further revisions you'd like. Please let me know if this direction is more aligned with what you had in mind.

JeanMeche commented:
Since this doesn't matches our expections, we close those suggested changes. We'll take care of it ourselves.
